go-controller/pkg/util/multi_network.go

```diff
@@ -595,8 +595,12 @@ func ParseNetConf(netattachdef *nettypes.NetworkAttachmentDefinition) (*ovncnity
 		return nil, fmt.Errorf("layer3 topology does not allow persistent IPs")
 	}
 
-	if netconf.PrimaryNetwork && netconf.Topology == types.LocalnetTopology {
-		return nil, fmt.Errorf("localnet topology does not allow primaryNetwork:true")
+	if netconf.Role != "" && netconf.Topology == types.LocalnetTopology {
+		return nil, fmt.Errorf("localnet topology does not allow network roles to be set since its always a secondary network")
```

Comment:
Is topology==localnet + role==secondary wrong ? 

Also we can improve the error message:

Let's first explain the error then the why.
Something like:
```
unexpected network field "role" for "localnet" topology, localnet topology does not allow network roles to be set since its always a secondary network
```

Reply:
basically for localnet its always secondary, so I'd rather err towards anybody setting this field for that topology as of now. we can easily add support for it later..

good point on error message, changes this